docs(flutter): Document `beforeCapture` function customization (#12190)
getsentry/sentry-docs: platform-includes/enriching-events/attach-screenshots/flutter.mdx

);
```

### Customize Screenshot Capturing

<Note>

Requires SDK version `8.11.0` or higher.

</Note>

> Because capturing screenshots can be a resource-intensive operation on Flutter, it's limited to one screenshot every 2 seconds using a debouncing mechanism. This behavior can be overruled if you supply a `BeforeCaptureCallback` for screenshots in the `SentryFlutterOptions`.

Reviewer: ```suggestion
Capturing screenshots on Flutter is limited to once every 2 seconds by default to minimize performance impact. While this debounce interval is fixed, you can override individual capture decisions by implementing the `beforeCaptureScreenshot` option in your `SentryFlutterOptions`.
```

The `BeforeCaptureCallback` also allows you to customize the behavior based on event data, so you can decide when to capture a screenshot and when not to. For example, you can decide to only capture screenshots of fatal events:

> ```java

Reviewer: ```suggestion
```dart
```

Reviewer: also please add line highlighting

Author: Not sure if highlighting makes sense here, as it's pretty much the whole code block.

await SentryFlutter.init((options) {

>   options.attachScreenshot = true;

Reviewer: ```suggestion
```

I guess we can remove this as we can assume screenshots are already in place

Reviewer: hmm there is a sample code above the beforeCapture though that shows how to enable screenshots

Author: Ah it's included, sry.
